http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -436,7 +436,14 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       .transaction
 
     res <- filterAppsBySamPermission(allClusters, userInfo, paramMap._3)
-  } yield res
+    // Make sure that the user still has access to the resource parent workspace
+    resWithWorkspace <- res.traverseFilter { resource =>
```

Comment:
would it make sense to check the users workspace permissions at the beginning (possibly even before querying) because the result of this `authProvider.isUserWorkspaceReader` would be the same for all apps? then just return nothing if the user doesn't have workspace permissions

Reply:
Oh yes, thanks for the catch! In my mind the list endpoint could eventually work across workspaces which is why I put it after, but no need to do that eight now while the endpoint still depends on having a workspace id (the fact that runtimes don't was confusing to me at first)

Reply:
oh gotcha, if you'd rather go with the more future-proof option I'm fine with that! with maybe a comment on that